resources/utils/maps/maps.go

```diff
@@ -0,0 +1,38 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package maps
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+func AsMapStringAny(item any) (map[string]any, error) {
```

Comment:
I get this method makes life easier, but it's quite non performant for something that should be simple.

If the map will have something around 7 or so entries not sure if this function gives real gains, it's not that painful to create such a map. Wdyt?

Reply:
It may not be that simple.  Those Azure SDK Properties structs that are converted using this function have a custom JSON marshaller and also they have inner layers. 

e.g. [BlobServicePropertiesProperties](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/storage/armstorage#BlobServicePropertiesProperties).

So we will have to both copy the field names and create all the inner `maps[string]any` as well.  It seemed a bit of dirty code that's why I followed this approach. 

Indeed is not performant there is a better solution to that, my first approach in this PRI was this:

```golang
func asMap(item any) (map[string]any, error) {
    reader, writer := io.Pipe()
    decoder := json.NewDecoder(reader)
    encoder := json.NewEncoder(writer)

    defer func() {
        _ = writer.Close()
        _ = reader.Close()
    }()

    go func() {
        _ = encoder.Encode(item)
    }()

    m := map[string]any{}
    err := decoder.Decode(&m)
    return m, err
}
```
That pipes the bytes from encoder to decoder and has less heap footprint. But then I thought it might be kind of complicated and changed it to marshal/unmarshal which Cloudbeat already uses elsewhere, with the intention to suggest the above as an improvement for the other parts as well.

Reply:
Do you need to always get all the properties?

Fine to keep like this too, just wondering if there is a better way to do it.